tools/manifest/sourcefile.py

```diff
@@ -212,54 +247,66 @@ def __getstate__(self):
         return rv
 
     def name_prefix(self, prefix):
+        # type: (bytes) -> bool
         """Check if the filename starts with a given prefix
 
         :param prefix: The prefix to check"""
         return self.name.startswith(prefix)
 
     def is_dir(self):
+        # type: () -> bool
         """Return whether this file represents a directory."""
         if self.contents is not None:
             return False
 
         return os.path.isdir(self.rel_path)
 
     def open(self):
+        # type: () -> BinaryIO
         """
         Return either
         * the contents specified in the constructor, if any;
         * a File object opened for reading the file contents.
         """
 
         if self.contents is not None:
-            file_obj = ContextManagerBytesIO(self.contents)
+            wrapped = ContextManagerBytesIO(self.contents)
+            if MYPY:
+                file_obj = cast(BinaryIO, ContextManagerBytesIO)
```

Comment:
```suggestion
                file_obj = cast(BinaryIO, wrapped)
```

if I understand how `cast` works from your other changes.